source/common/redis/proxy_filter.cc

```diff
@@ -19,9 +19,15 @@ ProxyFilterConfig::ProxyFilterConfig(const Json::Object& config, Upstream::Clust
       cluster_name_(config.getString("cluster_name")),
       stat_prefix_(fmt::format("redis.{}.", config.getString("stat_prefix"))),
       stats_(generateStats(stat_prefix_, scope)) {
-  if (!cm.get(cluster_name_)) {
-    throw EnvoyException(
-        fmt::format("redis filter config: unknown cluster name '{}'", cluster_name_));
+  Upstream::ThreadLocalCluster* cluster = cm.get(cluster_name_);
+  if (cluster == nullptr) {
+    throw EnvoyException(fmt::format("redis filter config: unknown cluster '{}'", cluster_name_));
+  }
+
+  if (cluster->info()->addedViaApi()) {
```

Comment:
I wonder if it's making things a bit messy to have to track this everywhere. The main use of it is to disallow registering Redis temporarily, and some assertions. As we head towards v2, everything is going to be dynamic, even if its sourced from a file. Is there a strong case to maintain this distinction across the call stack and objects?